### PR TITLE
 Validate the init command when it is executed on top of projects

### PR DESCRIPTION
**Pull request by NatashaWso2**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/11239

## Automation tests
 - Integration tests

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/InitCommand.java [hunk 1]
```diff
@@ -66,10 +74,37 @@ private static boolean isDirEmpty(final Path directory) throws IOException {
...
+            }
+            // If the current directory is not a project traverse and check down and up
+            if (!alreadyInitializedProject) {
+                // Recursively traverse down
+                Optional<Path> childDotBallerina = Files.walk(projectPath)
+                        .filter(path -> Files.isDirectory(path) &&
+                                path.toFile().getName().equals(ProjectDirConstants.DOT_BALLERINA_DIR_NAME))
+                        .findFirst();
+                if (childDotBallerina.isPresent()) {
+                    out.println("A ballerina project is already initialized in " + childDotBallerina.get().toFile()
```
Comment (manuranga) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/InitCommand.java]: `System.err` is more appropriate. WDYT?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/InitCommand.java [hunk 2]
```diff
@@ -66,10 +74,37 @@ private static boolean isDirEmpty(final Path directory) throws IOException {
...
+                            .getParent());
+                    return;
+                }
+                // Recursively traverse up till the root
+                Path projectRoot = findProjectRoot(projectPath);
+                if (projectRoot != null) {
+                    out.println("Directory is already within a ballerina project :" + projectRoot.toString());
```
Comment (manuranga) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/InitCommand.java]:
```suggestion
                    System.err.println("Directory is already within a ballerina project :" + projectRoot.toString());
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/InitCommand.java [hunk 3]
```diff
@@ -82,37 +117,50 @@ public void execute() {
...
-                    do {
-                        out.print("Version: (" + DEFAULT_VERSION + ") ");
-                        version = scanner.nextLine().trim();
-                        version = version.isEmpty() ? DEFAULT_VERSION : version;
-                    } while (!validateVersion(out, version));
-
-                    manifest.setVersion(version);
+                if (!manifestExistInProject) {
+                    // Check if Ballerina.toml file needs to be created.
+                    out.print("Create Ballerina.toml [yes/y, no/n]: (y) ");
```
Comment (manuranga) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/InitCommand.java]: it's move everything in this if to separate method, some sections go upto 7 levels of indentation

Comment (NatashaWso2) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/InitCommand.java]: Fixed with a46cc09

**Pull request merged**